android/app/src/main/java/com/chorestory/model/QuestRecyclerViewItem.java

```diff
@@ -9,38 +11,69 @@
 public class QuestRecyclerViewItem {
 
     private String name;
-    private int imageId;
     private int exp;
     private String owner;
     private String description;
     private boolean mandatory;
     private int dueDate;
+    private boolean recurring;
+    private int nextOccurrence;
 
-    public static List<QuestRecyclerViewItem> getData(QuestCompletion questCompletion) {
+    public static List<QuestRecyclerViewItem> getData(ArrayList<QuestParcelable> questParcelables) {
         List<QuestRecyclerViewItem> dataList = new ArrayList<>();
-        // TODO: fetch quests using QuestCompletion
-        for (int i = 0; i < 2; ++i) {
-            QuestRecyclerViewItem item = new QuestRecyclerViewItem("Sweep the floor", R.drawable.broom, 30, "Isabelle", "sweep it for 10 hours plz", true, 5);
+        for (int i = 0; i < questParcelables.size(); i++) {
+            QuestRecyclerViewItem item = new QuestRecyclerViewItem(questParcelables.get(i));
             dataList.add(item);
         }
         return dataList;
     }
 
-    QuestRecyclerViewItem(String name, int imageId, int exp, String owner, String description, boolean mandatory, int dueDate) {
-        this.name = name;
-        this.imageId = imageId;
-        this.exp = exp;
-        this.owner = owner;
-        this.description = description;
-        this.mandatory = mandatory;
-        this.dueDate = dueDate;
+    QuestRecyclerViewItem(QuestParcelable quest) {
+        this.name = quest.getTitle();
+        this.exp = quest.getReward();
+        this.owner = quest.getOwnerName();
+        this.description = quest.getDescription();
+        this.mandatory = true;
+        this.dueDate = quest.getDue();
+        this.recurring = quest.getReccurring();
+        this.nextOccurrence = quest.getNextOccurrence();
     }
 
     public String getName() {
         return name;
     }
 
+    private boolean stringContainsItemFromList(String inputStr, String[] items) {
+        for(int i =0; i < items.length; i++) {
+            if (inputStr.contains(items[i])) {
```

Comment:
let us make `inputStr` a Set of String instead, `inputStringSet`
to improve performance